cppimport/checksum.py

```diff
@@ -58,7 +51,8 @@ def checksum_save(module_data):
         module_data['extra_source_filepaths'] +
         [module_data['filepath']]
     )
-
     cur_checksum = calc_cur_checksum(dep_filepaths, module_data)
-    with open(checksum_filepath, 'wb') as file:
-        pickle.dump((dep_filepaths, cur_checksum), file)
+    dump = json.dumps([dep_filepaths, cur_checksum]).encode('ascii')
```

Comment:
Can you factor this into its own function, e.g. `append_checksum_trailer`?

Reply:
And add a comment/docstring that gives the explanation you provided in the PR description including linking to the post that suggests that this is legal?

Reply:
sure